core/models.py

```diff
@@ -369,6 +417,14 @@ class Meta:
     def __str__(self):
         return self.name
 
+    def get_id(cls, name, date):
+        return f"{name.lower().replace(' ', '_')}-{date.year}"
```

Comment:
If you're going to do it this way, set id to a https://docs.djangoproject.com/en/5.1/ref/models/fields/#generatedfield